lib/dat-worker-pool.rb

```diff
@@ -43,7 +43,7 @@ def start
   end
 
   def shutdown(timeout = nil)
-    @runner.shutdown(timeout)
+    @runner.shutdown(timeout){ caller }
```

Comment:
@jcredding why are you passing in the caller via a proc and not just a normal arg?  Are you wanting to delay execution of the `caller` statement to only call it if needed?